docker/bin/zookeeperStart.sh

```diff
@@ -31,101 +32,86 @@ fi
 
 MYID=$((ORD+1))
 
+# Values for first startup
 WRITE_CONFIGURATION=true
 REGISTER_NODE=true
+ONDISK_CONFIG=false
 
 # Check validity of on-disk configuration
 if [ -f $MYID_FILE ]; then
   EXISTING_ID="`cat $DATA_DIR/myid`"
-  if [[ "$EXISTING_ID" == "$MYID" || -f $DYNCONFIG ]]; then
-      WRITE_CONFIGURATION=false
+  if [[ "$EXISTING_ID" == "$MYID" && -f $STATIC_CONFIG ]]; then
```

Comment:
Variable `$DYNCONFIG` points to the first DynamicConfigFile "/data/zoo.cfg.dynamic" and its presence does not guarantee that dynamic configuration is present under "/data/conf/" directory Hence changed this check from existence check for "/data/conf/zoo.cfg" instead of "/data/zoo.cfg.dynamic".

Reply:
What's it that you are trying to validate here? A static config might exist independent of whether the server has ever joined the ensemble. I thought you're trying to validate that the server has participated in the ensemble previously.

Reply:
I'm just trying to validate if this is the first time we're bringing up this server.
In case of a server starting for the very first time, "/data/conf/zoo.cfg" (STATIC_CONFIG) will not exist, as it is copied later in the script to that location. See ln #109 in zookeeperStart.sh.
For first time server startup, we need to explicitly add the server to ensemble by setting REGISTER_NODE=true or incase of zk-0 by adding it directly as 'participant' to the ensemble.
On restarts server is not removed from ensemble, so it should exist in ensemble.